drain/drain.py

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env python3
+from lightning import Plugin, Millisatoshi, RpcError
+import re
+import time
+import uuid
+
+# TODOs:
+#  - fix: try increasing chunks on route errors
+#  - fix: sometimes, if we ran in error, not all chunk results are returned, i.e. [2/4, error] but not 1/4.
+#  - fix: occasionally strange route errors
+#  - feat: set HTLC_FEE MIN/MAX/STP by feerate
+#  - feat: convert a 'drain 0%' to a 'fill 100%' and vice versa instead of throwing error
+#  - chore: reconsider use of listchannels
```

Comment:
Might be worth putting these in a markdown README in the subdirectory. Makes it clearer that there are some TODOs and allows you to tick them off visibly :-)

Reply:
I will get this done in the next two weeks, I am about to have more time for all this... promise :D

Reply:
@cdecker  I want to change the meaning of `drain` and `fill` to not mean 'total target percentge' but rather just 'total percentage', as it caused confusions for @darosior for being unintuitive. Ignore over draining/filling (replacing it by whats possible). And add a `setbalance` command that does all the auto stuff.

Regarding the routing errors, I will have to check. Maybe we want this merged first.

Reply:
resolved by c8ac5cf `drain: document setbalance and todo`